testplan/common/entity/base.py

```diff
@@ -512,12 +513,17 @@ def generate_runpath(self):
         if self.parent and self.parent.runpath:
             return os.path.join(self.parent.runpath, self.uid())
```

Comment:
I remember for some derived classes, the uid() could just be its name which is a free style string. Do you meant to apply slugify on it, because slugify is imported but not used.